framework/src/base/NonlinearSystem.C

```diff
@@ -213,6 +214,45 @@ NonlinearSystem::stopSolve()
 void
 NonlinearSystem::setupFiniteDifferencedPreconditioner()
 {
+  std::shared_ptr<FiniteDifferencePreconditioner> fdp =
+      std::dynamic_pointer_cast<FiniteDifferencePreconditioner>(_preconditioner);
+  if (!fdp)
+    mooseError("Did not setup finite difference preconditioner, and please add a preconditioning "
+               "block with type = fdp");
+
+  if (fdp->finiteDifferenceType() == "coloring")
+    setupColoringFiniteDifferencedPreconditioner();
+  else if (fdp->finiteDifferenceType() == "standard")
+    setupStandardFiniteDifferencedPreconditioner();
```

Comment:
These methods are currently public, but you only call them internally. Do you want them public, protected, or private?